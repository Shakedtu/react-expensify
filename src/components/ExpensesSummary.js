import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import numeral from "numeral";
import selectExpensesTotal from "../selectors/expense-total";
import selectTotalFiltered from "../selectors/expense-total-filtered";
import selectExpenses from "../selectors/expenses";

const formattedExpense = (expensesTotal) => (
    numeral(expensesTotal / 100).format("$0,0.00"))

const expenseWording = (count) => count === 1 ? "expense" : "expenses";

export const ExpensesSummary = ({ expenseCount, expenseFilteredCount, expensesTotal, expenseFilteredTotal}) => {
    const expenseWord = expenseWording(expenseCount)
    const filteredWord = expenseWording(expenseFilteredCount)
    const formattedExpensesTotal = formattedExpense(expensesTotal);
    const formattedFilteredExpensesTotal = formattedExpense(expenseFilteredTotal);
    
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{expenseCount}</span> {expenseWord} totalling <span>{formattedExpensesTotal}</span></h1>
                <h3 className="page-header__title">Filtered <span>{expenseFilteredCount}</span> {filteredWord} totalling <span>{formattedFilteredExpensesTotal}</span></h3>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

const mapToStateProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    const unvisibleExpenses = selectTotalFiltered(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expenseFilteredCount: unvisibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses),
        expenseFilteredTotal: selectExpensesTotal(unvisibleExpenses)
    }
};

export default connect(mapToStateProps)(ExpensesSummary);