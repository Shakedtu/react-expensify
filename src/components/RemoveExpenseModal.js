import React from "react";
import Modal from "react-modal";


const RemoveExpenseModal = (props) => (
    <Modal
        className="modal"
        closeTimeoutMS={200}
        contentLabel="Are You Sure?"
        isOpen={props.isOpen}
    >
        <div className="modal-layout">
            <div className="modal-layout__box">
                <h3 className="modal-layout__title">Are you sure you want to delete this expense?</h3>
                <div className="modal-layout__buttons">
                    <button
                        className="button button--secondary"
                        onClick={props.onClickNo}
                    >
                        No
                    </button>
                    <button className="button" onClick={props.onClickYes}>Yes</button>
                </div>
            </div>
        </div>
    </Modal>
);

export default RemoveExpenseModal; 
