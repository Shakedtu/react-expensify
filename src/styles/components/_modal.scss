.ReactModalPortal > div {
    opacity: 0;
}

.ReactModalPortal .ReactModal__Overlay {
    align-items: center;
    display: flex;
    justify-content: center;
    transition: opacity 200ms ease-in-out;
}

.ReactModalPortal .ReactModal__Overlay--after-open {
    opacity: 1;
}

.ReactModalPortal .ReactModal__Overlay--before-close {
    opacity: 0;
}

.modal-layout {
    align-items: center;
    background: none;
    background-size: cover;
    display: flex;
    justify-content: center;
    height: 100vh;
    width: 100vw;
}

.modal-layout__box {
    background: fade-out($color: $off-white, $amount: 0.05);
    border-radius: 3px;
    padding: $l-size $m-size;
    text-align: center;
    width: 30rem;
}

.modal-layout__buttons {
    display: flex;
    justify-content: space-evenly;
}

.modal-layout__title {
    margin: 0 0 $m-size 0;
    line-height: 1.5;  
}