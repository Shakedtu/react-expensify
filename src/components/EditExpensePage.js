import React from "react";
import { connect } from "react-redux";
import ExpenseForm from "./ExpenseForm";
import { startRemoveExpense, startEditExpense } from "../actions/expenses";
import RemoveExpenseModal from "./RemoveExpenseModal";

export class EditExpensePage extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            isOpen: false
        };
    }

    onSubmit = (expense) => {
        this.props.startEditExpense(this.props.expense.id, expense);
        this.props.history.push("/");
    };

    onClickYes = () => {
        this.props.startRemoveExpense({ id: this.props.expense.id });
        this.props.history.push("/");
    };

    onClickNo = () => {
        this.setState(() => ({ isOpen: false }));
    };

    onClickRemove = () => {
        this.setState(() => ({ isOpen: true }));
    };

    render() {
        return (
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Edit Expense</h1>
                    </div>
                </div>
                <div className="content-container">
                    <ExpenseForm
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick={this.onClickRemove}>
                        Remove Expense
                    </button>
                </div>
                <RemoveExpenseModal
                    onClickYes={this.onClickYes}
                    onClickNo={this.onClickNo}
                    isOpen={this.state.isOpen}
                />
            </div>
        );
    }

}

const mapStateToProps = (state, props) => ({
    expense: state.expenses.find((expense) => (expense.id === props.match.params.id))
});

const mapDispatchToProps = (dispatch) => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
    startRemoveExpense: (data) => dispatch(startRemoveExpense(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);