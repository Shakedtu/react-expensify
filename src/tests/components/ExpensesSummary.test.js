import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("should correcly render ExpensesSummary with 1 expense", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={950}/>);
    expect(wrapper).toMatchSnapshot();

});

test("should correcly render ExpensesSummary with multiple expenses", () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={1100}/>);
    expect(wrapper).toMatchSnapshot();
});