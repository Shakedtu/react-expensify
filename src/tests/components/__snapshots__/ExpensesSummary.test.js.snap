// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correcly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $9.50
  </h1>
</div>
`;

exports[`should correcly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $11.00
  </h1>
</div>
`;
